File: solutions/ruby/raindrops/1/raindrops.rb
class Raindrops
  VERSION = 1

  @@sounds = { 3 => 'Pling', 5 => 'Plang', 7 => 'Plong' }

  def self.convert(n)
    result = ''
    @@sounds.map { |k, v| result += v if n % k == 0 }
    result += n.to_s if result.empty?
    result
  end
end

File: solutions/haskell/raindrops/1/src/Raindrops.hs
module Raindrops (convert) where

convert :: Int -> String
convert n
    | null sounds = show n
    | otherwise   = sounds
    where
        sounds = concatMap ifFactor [(3, "Pling"), (5, "Plang"), (7, "Plong")]
        ifFactor (f, str) = if n `rem` f == 0 then str else ""